### ISSUE TITLE
subplot() fails to plot 6 plots in a 2 rows x 4 columns grid

### ISSUE THREAD

**Issue by louievdl** [plotly/plotly.R#2400]
I want to use _subplot_ function to plot 6 plots in a 2 rows x 4 columns grid.

Here is a reproducible example:
```
figlist <- list()
for (i in 1:6) figlist[[i]] <- ggplot(data.frame(a = 1:3,b = 4:6), aes(a,b)) + geom_point()
subplot(figlist, nrows = 2, widths = rep(0.25,4))
```

This generates "Error: The length of the widths argument must be equal to the number of columns"

Looking into the code of subplots.R, the _subplot_ function accepts arguments _nrows_, _widths_ and _heights_, which appear to be passed through to _get_domains_.

The problem is that _get_domains_ calculates _ncols_ ignoring _length(widths)_ and then generates an error when these don't match.

I would have expected _get_domains_ to calculate _ncols_ only if _widths_ was not supplied.

I hope this helps.

**Comment by romanzenka** [comment]
I do not think I would want plotly to work the way you envision. 

A more reasonable approach would be to allow the user to set both nrows and ncols, and be able to generate "blank spaces" when there are not enough plots to fill all the slots.

Then if you say ncols=4 and provide 4 widths, it should work.

**Comment by louievdl** [comment]
Yes, as you say, probably better to add an ncols option for subplot - if the user didn't provide widths, subplot could just assume equal widths.
